pex/resolve/pep_691/api.py

```diff
@@ -0,0 +1,197 @@
+                        path=path, key=key
+                    )
+                )
+
+            if not isinstance(value, expected_type):
+                raise response_error(
+                    "was expected to contain a {expected_type} at '{path}[\"{key}\"]' but "
+                    "contained a {type} instead.".format(
+                        expected_type=expected_type.__name__,
+                        path=path,
+                        key=key,
+                        type=type(value).__name__,
+                    )
+                )
+
+            return value
+
+        api_version = Version(get("api-version", text, obj=get("meta", dict), path=".meta"))
+        if not isinstance(api_version.parsed_version, PackagingVersion):
+            raise response_error(
+                "reports an api-version of {api_version} which does not have the required "
+                "<major>.<minor> structure.".format(api_version=api_version.raw)
+            )
+        if api_version.parsed_version.major != 1:
+            raise response_error(
+                "reports an api-version of {api_version} and Pex currently only supports "
+                "api-version 1.x".format(api_version=api_version.raw)
+            )
+
+        name = ProjectName(get("name", text))
+
+        files = []
+        for index, file in enumerate(get("files", list, default=[]), start=0):
+            path = ".files[{index}]".format(index=index)
+
+            fingerprints = []
+            for algorithm, hash_ in get("hashes", dict, obj=file, path=path).items():
+                if not isinstance(hash_, text):
+                    raise response_error(
+                        "reports a hash value of {hash} of type {type} for "
+                        "'{path}.hashes[\"{algorithm}\"]' but hash values should be strings".format(
+                            hash=hash_, type=type(hash_).__name__, path=path, algorithm=algorithm
+                        )
+                    )
+                fingerprints.append(
+                    Fingerprint(
+                        algorithm=algorithm,
+                        # N.B.: Hash values are hex which is ascii; so cast(str, hash_) is correct
+                        # for Python 2.7.
+                        hash=cast(str, hash_),
+                    )
+                )
+
+            files.append(
+                File(
+                    filename=get("filename", text, obj=file, path=path),
+                    # N.B.: All URLs in the PEP-691 API are allowed to be relative, see:
+                    #   https://peps.python.org/pep-0691/#json-serialization
+                    # The `urljoin` functions does the right thing here and creates an absolute URL
+                    # only when needed (never for the current PyPI scheme, potentially for other
+                    # indexes though).
+                    url=urlparse.urljoin(endpoint.url, get("url", text, obj=file, path=path)),
```

Comment:
Maybe worth a dedicated unit test? This seems niche and unlikely to be encountered frequently in the wild, so it would be easy to regress here without realizing.

Reply:
Well, it won't be easy to regress since it is tested already, but I think breaking out the test to a dedicated unit test makes sense for other reasons.

Reply:
Done.